cli/chaosli/cmd/context.go

```diff
@@ -0,0 +1,393 @@
+	// do not require namespaces to be described
+	podNamespaces := disruption.ObjectMeta.Namespace
+	rowNames := 1
+
+	if level == "node" {
+		rowNames = 1
+	}
+
+	fmt.Println("Let's look at your targets...")
+
+	s := spinner.New(spinner.CharSets[38], 100*time.Millisecond)
+	s.Start()
+	cmd := fmt.Sprintf("kubectl get %s -n %s -l %s | wc -l", level, podNamespaces, labels)
+	sizeString, err := exec.Command("bash", "-c", cmd).Output()
+
+	if err != nil {
+		return nil, fmt.Errorf("Could not count the number of pods correlating to target selector: %v", err)
+	}
+	size, err := strconv.Atoi(strings.Trim(string(sizeString), "\n"))
+
+	if err != nil {
+		return nil, fmt.Errorf("Could not convert string to integer in context target size: %v", err)
+	}
+
+	// Remove header NAME from consideration
+	size = size - 1
+	// If the size is 0, first check if changing the level will do anything, otherwise
+	// mention to the user that the labels they are using won't target anything
+
+	if size <= 0 {
+		errorString := fmt.Sprintf("\nThe label selectors chosen (%s) result in 0 targets, meaning this disruption would do nothing given the namespace/cluster/label combination.", labels)
+		newLevel := ""
+
+		if level == POD {
+			newLevel = "node"
+		} else {
+			newLevel = POD
+		}
+		cmd := fmt.Sprintf("kubectl get %s -n %s -l %s | wc -l", newLevel, podNamespaces, labels)
+		sizeString, err := exec.Command("bash", "-c", cmd).Output()
+
+		if err != nil {
+			return nil, fmt.Errorf("Could not count the number of pods correlating to target selector: %v", err)
+		}
+		size, err = strconv.Atoi(strings.Trim(string(sizeString), "\n"))
+
+		if err != nil {
+			return nil, fmt.Errorf("Could not convert string to integer in context target size: %v", err)
+		}
+
+		if size > 0 {
+			errorString = fmt.Sprintf("\nWe noticed that your target size is 0 for level %s given your label selectors. We checked to see if the %s level would give you results and we found %d %ss. Is this the level you wanted to use?", level, newLevel, size, newLevel)
+		}
+		fmt.Println(errorString)
+		return nil, fmt.Errorf("bad sized targets")
+	}
+
+	cmd = fmt.Sprintf("kubectl get %s -n %s -l %s | awk '{print $%d}'", level, podNamespaces, labels, rowNames)
+	targets, err := exec.Command("bash", "-c", cmd).Output()
+
+	if err != nil {
+		return nil, fmt.Errorf("Could not grab list of targets names correlating to target selector: %v", err)
+	}
+
+	s.Stop()
+
+	targetsShow := []string{}
+	targetsAll := []string{}
+	targetsSplit := strings.Split(string(targets), "\n")
+
+	for i := 0; i < len(targetsSplit); i++ {
+		if len(targetsShow) < MAXTARGETSHOW {
+			targetsShow = append(targetsShow, targetsSplit[i])
+		}
+		if targetsSplit[i] == "NAME" || targetsSplit[i] == "" {
+			continue
+		}
+		targetsAll = append(targetsAll, targetsSplit[i])
+	}
+
+	fmt.Printf("\n🎯 There are %d %ss that will be targeted\n", size, level)
+	fmt.Println("📝 Here are a small set of those targets:")
```

Comment:
Should this message be dynamic? Sometimes (probably often, even), we'll be showing all of the targets

Reply:
Dynamic how so?

Reply:
As in only printing this string if we really have truncated the list

Reply:
Oh right, good point.